index.js

```diff
@@ -46,14 +51,18 @@ function sanitize(input, replacement) {
     .replace(reservedRe, replacement)
     .replace(windowsReservedRe, replacement)
     .replace(windowsTrailingRe, replacement);
-  return truncate(sanitized, 255);
+  return preserveFileExt ? 
```

Comment:
Hmm.. I think we shouldn't touch the extension while truncating. In my view, the algorithm here would be something like:

- split out the extension, if any
- subtract the length of the extension (including the dot) from `truncateLength`. If there's no extension, then truncate the entire string
- if greater, truncate the filename (without the dot and extension) to the length calculated in the previous step
- re-join the truncated filename with the extension

As a bonus, it would be nice to only truncate if the length of the filename exceeds `truncaeLength` in both cases